index.js

```diff
@@ -21,8 +24,29 @@ const airtableUrl = config.airtableUrl.trim();
 const encodedCredentials = Buffer.from(`${clientId}:${clientSecret}`).toString('base64');
 const authorizationHeader = `Basic ${encodedCredentials}`;
 
+// book keeping to using this easier, not needed in a real implementation
+setLatestTokenRequestState('NONE');
+
 app.get('/', (req, res) => {
-    res.send('<a href="redirect-testing">Testify!</a>');
+    const latestRequestStateDisplayData = formatLatestTokenRequestStateForDeveloper();
+    res.send(`
+  <div>
+    <h3> New Token</h3>
+    <a href="redirect-testing">Click to authorize and create a new access token</a>
+    <br/>
+    <h3>Refresh a token</h3>
+    ${latestRequestStateDisplayData}
+    <p>
+        To test refreshing a token, enter it into the input and press "submit"
+        <br/>
+        In your own code, refreshing should occur as a background process.
+    </p>
+    <form action="/refresh_token" method="post" >
+        <label for="refresh">Refresh token:
+        <input type="text" id="refresh" name="refresh_token" autocomplete="off" minLength="64"/>
+        <input type="submit">
```

Comment:
are we worried about submit being doubleclicked and the token being invalidated due to double usage of the same token?

Reply:
Left a comment, don't think it's worth fixing though